src/cmap/connection.ts

```diff
@@ -24,30 +32,60 @@ const kDescription = Symbol('description');
 const kIsMaster = Symbol('ismaster');
 const kAutoEncrypter = Symbol('autoEncrypter');
 
-class Connection extends EventEmitter {
-  id: any;
-  address: any;
-  socketTimeout: any;
-  monitorCommands: any;
-  closed: any;
-  destroyed: any;
-  [kDescription]: any;
-  [kGeneration]: any;
-  [kLastUseTime]: any;
-  [kAutoEncrypter]: any;
-  [kQueue]: any;
-  [kMessageStream]: any;
-  [kStream]: any;
-  [kIsMaster]: any;
-  [kClusterTime]: any;
-
-  constructor(stream: any, options: any) {
+export interface MongoDBConnectionOptions
```

Comment:
I thought replied to this but I think it got lost, 
'tls' exports a ConnectionOptions type, which I do rename, you can see I do `import type { ConnectionOptions as TLSConnectionOptions } from 'tls';` but I wonder if we ever export this type we'd want to avoid that clash, right?

Reply:
hm I don't know if we'd ever want to export this type. I'd prefer to cross that bridge when we come to it, but for now `MongoDBConnectionOptions` sticks out like a sore thumb since nothing else is named with the same prefix